VOID/dockers/success.py

```diff
@@ -40,6 +44,61 @@ def dock(self, attempts):
 
             if self.fitness(cpx) >= 0:
                 print(f"{trial + 1} attempts to success")
+                cpx = self.rescale(cpx)
                 return [cpx]
 
         return []
+
+    def rescale(self, complex):
+        """Rescale the complex to the 0-1 range so results can be visualized in direct and xyz format.
+
+        Args:
+            complex (Complex): The host-guest complex object containing the host and guest molecules.
+
+        Returns:
+            Complex: The rescaled host-guest complex object.
+        """
+        lattice = complex.pose.lattice
+        frac_coords = []
+        species_list = []
+        site_labels = []
+
+        for site in complex.pose:
+            site_labels.append(site.label)
+            species_list.append(site.species)
+            coords = (
+                site.frac_coords
+                if site.label == "host"
+                else np.mod(site.frac_coords, 1.0)
+            )
+            frac_coords.append(coords)
+
+        site_properties = {"label": site_labels}
+
+        updated_structure = Structure(
+            lattice, species_list, frac_coords, site_properties=site_properties
+        )
+
+        num_host_atoms = len(complex.host)
+
+        species = updated_structure.species
+        cart_coords = updated_structure.cart_coords
+
+        # Update the host and guest with the rescaled 0-1 species and coordinates
+        complex.host = Structure(
+            lattice=complex.host.lattice,
+            species=species[:num_host_atoms],
+            coords=cart_coords[:num_host_atoms],
+            coords_are_cartesian=True,
+            site_properties=complex.host.site_properties,
+        )
+
+        complex.guest = Structure(
+            lattice=complex.host.lattice,
+            species=species[num_host_atoms:],
+            coords=cart_coords[num_host_atoms:],
+            coords_are_cartesian=True,
+            site_properties=complex.guest.site_properties,
+        )
```

Comment:
This code might not follow best practices: normally, you don't want to modify the inputs to a function. It's not as efficient, but it would be better to create a copy of the input, modify the copy, and return the copy on the last line.

Reply:
I'm not sure that I completely get what you are saying. You mean substituting the `self.rescale(cpx)` in dock by the whole process inside the function? or ammending the rescale function so it does not modify the input cpx? 
(I'm mostly sure you mean option B, but better double check)

```

    def dock(self, attempts):
        cpx = Complex(self.host.copy(), self.guest.copy())

        for trial in range(attempts):
            cpx = self.run(cpx, 1)

            if self.fitness(cpx) >= 0:
                print(f"{trial + 1} attempts to success")
                cpx = self.rescale(cpx)
                return [cpx]

        return []
```

Reply:
I did mean the second thing you asked. Right now, the `rescale` function you wrote modifies the input `complex`. You have something like this:
```python
def rescale(self, complex):
    ...
    complex.host = Structure(...)
    ...
    return complex
```

Best coding practices suggest that you should make a copy of the input and modify that instead, like this:
```python
def rescale(self, complex):
    cpx = complex.copy() # this might not work for Complex objects, you should be able to use copy.deepcopy() as an alternative
    cpx.host = Structure(...)
    return cpx
```

Reply:
Cool, I'll ammend this and give it a test

Reply:
At the end .copy.deepcopy() did not work and .copy() itself worked fine when testing.
Closing this PR and opening another with the changes made incorporated